consent/strategy_default.go

```diff
@@ -190,13 +190,7 @@ func (s *DefaultStrategy) getIDTokenHintClaims(ctx context.Context, idTokenHint
 	} else if err != nil {
```

Comment:
What about line 188?

```
if ve := new(jwtgo.ValidationError); errors.As(err, &ve)
```

I do not think this works because new `ValidationError` does not implement [`causer` interface](https://github.com/pkg/errors#retrieving-the-cause-of-an-error) anymore? So I do not think `errors.As` really works here?

Reply:
`errors.As` does not depends on `causer` interface, see [source code here](https://golang.org/src/errors/wrap.go?s=2494:2537#L67). It works because `Decode` returns `ValidationError`. `errors.As` depends on `As` interface and `Unwrap` interface. It first checks if the root error can be assigned to the target interface, which is our case, then it checks if the error implements `As` interface to outsource the check, then if the error can be `Unwrapp`ed it repeats the process with the Unwrapped error.

https://play.golang.org/p/yXkD0I0fG4F

Reply:
Oh, you are right, I missed that `errors.As` just calls into stdlib: https://github.com/pkg/errors/blob/master/go113.go#L31